src/api/courses.js

```diff
@@ -4,13 +4,87 @@ const router = require('express').Router({
 
 const { check } = require('express-validator/check')
 const { matchedData } = require('express-validator/filter')
+const moment = require('moment')
 
-const { Course, Queue, User } = require('../models')
+const { Course, Queue, Question, User, Sequelize } = require('../models')
 const { requireCourse, requireUser, failIfErrors } = require('./util')
 const requireAdmin = require('../middleware/requireAdmin')
 const requireCourseStaff = require('../middleware/requireCourseStaff')
 const safeAsync = require('../middleware/safeAsync')
 
+// Helper for getting course question data
+const getColumns = questions => {
+  const columns = new Set()
+  const selectedColumns = new Set([
+    'id',
+    'topic',
+    'enqueueTime',
+    'dequeueTime',
+    'answerStartTime',
+    'answerFinishTime',
+    'comments',
+    'preparedness',
+    'askeBy.UserLocation',
+    'answeredBy.AnsweredBy_netid',
+    'answeredBy.AnsweredBy_UniversityName',
+    'askedBy.AskedBy_netid',
+    'askedBy.AskedBy_UniversityName',
+    'queue.queueId',
+    'queue.courseId',
+    'queue.QueueName',
+    'queue.QueueLocation',
+    'queue.Queue_CreatedAt',
+    'queue.course.CourseName',
+  ])
+
+  questions.forEach(question => {
+    Object.keys(question).forEach(questionKey => {
+      if (selectedColumns.has(questionKey)) {
+        columns.add(questionKey)
+      }
+    })
+  })
+  return columns
+}
+
+const getCsv = questions => {
+  const columns = getColumns(questions)
+  // Taken from https://stackoverflow.com/questions/8847766/how-to-convert-json-to-csv-format-and-store-in-a-variable
+  const timeFields = new Set([
+    'queue.Queue_CreatedAt',
+    'enqueueTime',
+    'dequeueTime',
+    'answerStartTime',
+    'answerFinishTime',
+  ])
+  const header = Array.from(columns)
+  const replacer = (key, value) => (value === null ? '' : value)
+  const csv = questions.map(row =>
+    header
+      .map(fieldName => {
+        if (timeFields.has(fieldName)) {
+          const time = row[fieldName]
+          const formattedTime =
+            time !== null
+              ? moment
+                  .tz(time, 'YYYY-MM-DD HH:mm:ss.SSS Z', 'US/Central')
```

Comment:
Thought: this may be undesirable if/when we roll out to schools like UBC that aren't in the central timezone. When that happens, we should either a) report times in UTC and allow consumers of this data to transform it as needed, or b) associate some concept of timezone with either courses or institutions (one institutions are an entity that we have). I think I'd prefer the latter; at any rate, that's outside the scope of this PR.